trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/SifaRunner.java

```diff
@@ -82,18 +82,17 @@ public final class SifaRunner<L extends IIcfgTransition<?>> implements IInterpol
 	private final InterpolantComputationStatus mInterpolantComputationStatus;
 
 	public SifaRunner(final IUltimateServiceProvider services, final ILogger logger, final IIcfg<?> icfg,
-			final IRun<L, ?> currentCex, final IPredicateUnifier unifier) {
-		mTrace = currentCex.getWord().asList();
+			final Word<L> currentCex, final IPredicateUnifier unifier) {
+		mTrace = currentCex.asList();
 		mPredicateUnifier = unifier;
 		mPrecondition = mPredicateUnifier.getTruePredicate();
-		final Set<L> pathProgramSet = currentCex.getWord().asSet();
+		final Set<L> pathProgramSet = currentCex.asSet();
 		final PathProgramConstructionResult ppConstructionResult = PathProgram.constructPathProgram("sifa-path-program",
 				icfg, pathProgramSet, Collections.emptySet(), x -> true);
 		final PathProgram pp = ppConstructionResult.getPathProgram();
 
-		final List<IcfgLocation> locationOfInterestList =
-				TraceCheckUtils.getSequenceOfProgramPoints(currentCex.getWord()).stream()
-						.map(a -> ppConstructionResult.getLocationMapping().get(a)).collect(Collectors.toList());
+		final List<IcfgLocation> locationOfInterestList = TraceCheckUtils.getSequenceOfProgramPoints(currentCex)
+				.stream().map(a -> ppConstructionResult.getLocationMapping().get(a)).collect(Collectors.toList());
```

Comment:
I do not know anything about our SIFA implementation, so I tried to leave it as it is. In particular, this class does something more complex with `locationOfInterestList`, and I do not know how to map that.

As SIFA so far only works for sequential programs generally, I think generalizing this is not high-priority.

Reply:
Nevertheless, for consistency, it should use the sequence of control locations (although it will fail for any sequence that is different from a sequence of program locations).

Reply:
This would require some changes in the callers (the CEGAR loops). Right now, `NwaCegarLoop` always passes a `List<Set<IcfgLocation>>` for the control configurations. We would have to change it such that
- for sequential programs, it passes a `List<IcfgLocation>`;
- and for concurrent programs, it passes a `List<Set<IcfgLocation>>` or `List<IcfgLocation[]>`

I'll look into it. Then we could also perhaps restructure these two similar methods in `NwaCegarLoop` that you remarked on.

Reply:
On second thought [first reponse deleted], I do not think that is a good idea. It will break the contract that the control configurations in a `Counterexample<L>` are only used in equality comparisons.

Reply:
It might be worthwhile to think about this use case. 

I don't see why we restricting ourselves to `Object` in `Counterexample<L>`. What is preventing us from having a `Counterexample<L,C>`  where `C` is the type of control location? 

Nevertheless, it is fine here either way.